code/src/actors/link_puppet.c

```diff
@@ -0,0 +1,147 @@
+#include <stddef.h>
+#include <string.h>
+
+#include "link_puppet.h"
+#include "common.h"
+
+// Draw at most three extra Links. More can cause graphical issues that persist until game restart.
+#define MAX_PUPPETS_AT_ONCE 3
+
+ActorInit EnLinkPuppet_InitVars = {
+    0x1,                  // ID
+    ACTORTYPE_NPC,        // Type
+    0xFF,                 // Room
+    0b110000,             // Flags
+    21,                   // Object ID (20: Adult, 21: Child)
+    sizeof(EnLinkPuppet), //
+    EnLinkPuppet_Init,    //
+    EnLinkPuppet_Destroy, //
+    EnLinkPuppet_Update,  //
+    EnLinkPuppet_Draw,    //
+};
+
+// When posing the model by copying the joint table, the model for some reason gets raised about 12 units.
+u32 ChildYPosOffset() {
+    return playingOnCitra ? 12 : 0;
+}
+
+typedef void (*SkelAnime_InitLink_proc)(SkelAnime* skelAnime, ZARInfo* zarInfo, GlobalContext* globalCtx, void* cmbMan,
+                                        void* param_5, u32 animation, s32 limbBufCount, void* jointTable,
+                                        void* morphTable);
+#define SkelAnime_InitLink ((SkelAnime_InitLink_proc)0x3413EC)
+
+void EnLinkPuppet_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnLinkPuppet* this = (EnLinkPuppet*)thisx;
+
+    this->base.room = -1;
+
+    SkelAnime_InitLink(&this->skelAnime, PLAYER->zarInfo, globalCtx, PLAYER->cmbMan, thisx->unk_178, 0, 9, NULL, NULL);
```

Comment:
I think the custom tunic glitches would be fixed if, after this call to `SkelAnime_InitLink`, you do something similar to what is done in `Player_SetChildCustomTunicCMAB`.
The only things to change should be the Texanim value (to match Link's current age) and replacing `PLAYER->skelAnime` with `this->skelAnime`.

Reply:
Thanks!